peer.go

```diff
@@ -75,52 +75,57 @@ func (id ID) ExtractEd25519PublicKey() ic.PubKey {
 	// Decode multihash
 	decoded, err := mh.Decode([]byte(id))
 	if err != nil {
-		// This error should not occur for any identity
-		log.Error("Unable to decode multihash", id)
-		return nilPubKey
+		return nilPubKey, fmt.Errorf("Unable to decode multihash")
 	}
 
 	// Check ID multihash codec
 	if decoded.Code != mh.ID {
-		return nilPubKey
+		return nilPubKey, fmt.Errorf("Unexpected multihash codec")
```

Comment:
Three things:
1. errors should be lower case
2. for static errors it is better to use `errors.New`
3. might be nice to define them before as public variable so they can be compared with